### PR TITLE
✨ add Week5

### PR DESCRIPTION
**Pull request by LouisaStrunk02**
Changes:

1. add create_family and free_family functions to inheritance
2. add dictionary.c to speller

### CODE REVIEW

File: week5/speller/dictionary.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+node *table[COLUMNS];
+
+bool check(const char *word)
+{
+    int hash_value = hash(word);
+    node *cursor = table[hash_value];
+
+    while (cursor != NULL)
+    {
+        bool wordIsInDictionary = (strcasecmp(cursor->word, word) == 0);
```
Comment (stefanluth) [inline week5/speller/dictionary.c:28]: You used [snake_case](https://en.wikipedia.org/wiki/Snake_case) above and below :)

File: week5/speller/dictionary.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+{
+    FILE *file = fopen(dictionary, "r");
+
+    if (file == NULL)
+    {
+        printf("Could not open file");
+
+        return false;
+    }
+    char new_word[LENGTH + 1];
```
Comment (stefanluth) [inline week5/speller/dictionary.c:57]:
```suggestion
    }

    char new_word[LENGTH + 1];
```

File: week5/speller/dictionary.c [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            return false;
+        }
+
+        strcpy(new_node->word, new_word);
+        unsigned int index = hash(new_word);
+        new_node->next = table[index];
+        table[index] = new_node;
+        word_counter++;
+    }
+    fclose(file);
```
Comment (stefanluth) [inline week5/speller/dictionary.c:73]:
```suggestion
    }

    fclose(file);
```

File: week5/speller/dictionary.c [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        node *cursor = table[i];
+
+        while (cursor)
+        {
+            node *temp = cursor;
+            cursor = cursor->next;
+            free(temp);
+        }
+        if (cursor == NULL)
```
Comment (stefanluth) [inline week5/speller/dictionary.c:95]:
```suggestion
        }

        if (cursor == NULL)
```

**Review by stefanluth — APPROVED** [review]

**Comment by stefanluth** [comment]
Looks good :)

**Review by shoreal — APPROVED** [review]

**Pull request merged**